### PR TITLE
Eri9600/densify and generalize

### PR DESCRIPTION
**Pull request by eri9600**
Densify and Generalize sample as described here: https://devtopia.esri.com/runtime/common-samples/issues/202

### CODE REVIEW

**Comment by eri9600** [comment]
Hi @TADraeseke, 
Please review when time permits.
Thanks!

File: java/densify-and-generalize/src/main/java/com/esri/arcgisruntime/sample/densifygeneralize/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    Multipoint multipoint = new Multipoint(tempPolyline.getParts().getPartsAsPoints());
+    mResultPointGraphic.setGeometry(multipoint);
+  }
+
+  /**
+   * Creates a collection of points along the Willamette River in Portland, OR.
+   *
+   * @return points
+   */
+  private static PointCollection createShipPoints() {
```
Comment (TADraeseke) [inline java/densify-and-generalize/src/main/java/com/esri/arcgisruntime/sample/densifygeneralize/MainActivity.java]: Why are these called Ship points? I wouldn't want to sail a ship on a generalized line...

Anyways, it looks like all the other platforms have already called it that. I guess the 'ship has sailed' on this :P

Comment (eri9600) [reply java/densify-and-generalize/src/main/java/com/esri/arcgisruntime/sample/densifygeneralize/MainActivity.java]: I agree that the two checkboxes don't really do much to the sample. I was following what Java did. Should I remove them ?

Comment (TADraeseke) [reply java/densify-and-generalize/src/main/java/com/esri/arcgisruntime/sample/densifygeneralize/MainActivity.java]: Lets leave them in if they're implemented in other samples.

File: java/densify-and-generalize/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## How it works
+To densify and generalize a polyline 
+1. Use the static method `GeometryEngine.densify(polyline, maxSegmentLength)` to densify the polyline . The resulting polyline will add points along the line so that there are no points greater than `maxSegmentLength` from the next point.
+2. Use the static method `GeometryEngine.generalize(polyline, maxDeviation, true)` to generalize the polyline. The resulting polyline will have points or shifted from the line to simplify the shape. None of these points can deviate farther from the original line than `maxDeviation`. The last parameter, `removeDegenerateParts` , will clean up extraneous parts if the geometry is multi-part it will have no effect in this sample.
+3. Note that `maxSegmentLength` and `maxDeviation` are in the units of geometry's coordinate system. This could be in degrees in some coordinate systems. In this example, a cartesian coordinate system is used and at a small enough scale that geodesic distances are not required.
+
+## Features
```
Comment (TADraeseke) [inline java/densify-and-generalize/README.md]: Relevant API

**Comment by eri9600** [comment]
@TADraeseke i merged the sample into master. Thanks!

**Pull request merged**